### ISSUE TITLE
Very urgent: An error occurred in initiatePayment when creating a Stripe customer:\r\nundefined\r\n | "secret_key_required"

### ISSUE THREAD

**Issue by Sylvesterx1** [medusajs/medusa#5583]
Medusa full App  ( admin + storefront)

**Plugin used**: medusa-stripe-payment

Not possible select payment by Stripe example: during select credit card POST return 400:
```
{code: "secret_key_required", type: "invalid_data",…}
code
: 
"secret_key_required"
message
: 
"An error occurred in initiatePayment when creating a Stripe customer:\r\nundefined\r\n"
type
: 
"invalid_data"
```

I read all documentations related to Plugin Medusa Stripe, and each guide shown only information about add public key.

backend:
`.env ADDED: STRIPE_API_KEY=pk_live_51M....`

strorefront.
`.env.local  ADDED NEXT_PUBLIC_STRIPE_KEY=pk_live_51M...`

medusa-config.js:
```
{
    resolve: `medusa-payment-stripe`,
    options: {
      api_key: process.env.STRIPE_API_KEY,
      webhook_secret: process.env.STRIPE_WEBHOOK_SECRET,
    },
  },
```

Still retrive above error. For test I try generate secret key and initialize:
```
{
    resolve: `medusa-payment-stripe`,
    options: {
      api_key: process.env.STRIPE_API_KEY,
	secret_key: process.env.STRIPE_SECRET_KEY,
      webhook_secret: process.env.STRIPE_WEBHOOK_SECRET,
    },
  },
```

Finally is backend admin
.env
```
STRIPE_API_KEY=pk_live_51M....
STRIPE_WEBHOOK_SECRET=whsec_vg...
STRIPE_SECRET_KEY=sk_live_51MkTi...
```

storefront
.env.local
```
# Your Stripe public key. See – https://docs.medusajs.com/add-plugins/stripe
NEXT_PUBLIC_STRIPE_KEY=pk_live_51M.....
(ADDEDED)NEXT_SECRET_STRIPE_KEY=sk_live_51MkT....
```

Still: `{code: "secret_key_required", type: "invalid_data",…}`

Im not sure this plugin correct read my line which I add  for storefront env with secret:
`NEXT_SECRET_STRIPE_KEY=sk_live_51MkT....`

Maybe there should be another variable name for secret? Thanks for verifying.

**Comment by Sylvesterx1** [comment]
@MedusaNick 
@Medusa Admins,
Is possible provide more information? Because every documentation states that there is full integration with Stripe Payment with storefront,  and now I don't know where this error may come from. This is the last stage before launching the store. Thank you for verifying.

**Comment by olivermrbl** [comment]
@Sylvesterx1, looking at [Stripe's official documentation](https://stripe.com/docs/error-codes#:~:text=secret_key_required,your%20integration%20to%20use%20them.), this error is thrown when you are using a publishable API key where you should be using a secret key.

Can I get you to ensure you are using a secret key in the Medusa plugin configuration?

**Comment by Sylvesterx1** [comment]
@olivermrbl Typically, in every e-commerce system for Stripe API integration, we use two:

- Stripe Public API key
- Stripe Secret API key

For default Medusa in .env configuration require only (also all videos and documentation show to place here PUBLIC API KEY):
```
# Your Stripe public key. See – https://docs.medusajs.com/add-plugins/stripe
NEXT_PUBLIC_STRIPE_KEY=MY_PUBLIC_KEY
```

When we place in this  .env Stripe Public Key  and during select Stripe Credit Cart in checkout then we get:
```
Not possible select payment by Stripe example: during select credit card POST return 400:

{code: "secret_key_required", type: "invalid_data",…}
code
: 
"secret_key_required"
message
: 
"An error occurred in initiatePayment when creating a Stripe customer:\r\nundefined\r\n"
type
: 
"invalid_data"

```

and now fallowing to your instruction for test  I make changes  in .env storefront file and in the same place i remove Public Key and add Stripe SECRET KEY.

```
# Your Stripe public key. See – https://docs.medusajs.com/add-plugins/stripe
NEXT_PUBLIC_STRIPE_KEY=MY_SECRET_KEY_HERE
```


Now I can select payment Stripe and  I see form for fill credit card details, also I see in Stripe account payment initializaon request,  but is still displayed error in app because not possible write card data in form. 

I checked in log console and:
```
v3:1 Uncaught (in promise) IntegrationError: You should not use your secret key with Stripe.js.
          Please pass a publishable key instead.
    at l (v3:1:140604)
    at new e (v3:1:364802)
    at Fc (v3:1:403657)
    at initStripe (webpack-internal:///(:8000/app-pages-browser)/./node_modules/@stripe/stripe-js/dist/stripe.esm.js:105:28)
    at eval (webpack-internal:///(:8000/app-pages-browser)/./node_modules/@stripe/stripe-js/dist/stripe.esm.js:129:12)
```

**Comment by kasperkristensen** [comment]
@Sylvesterx1 from your initial message it looks like you have added the publishable key (`pk_...`) to both the backend and storefront. You need to add the secret key to your backend project `STRIPE_API_KEY=sk_...` and the public key to your storefront `NEXT_PUBLIC_STRIPE_KEY=pk_...`. The setup is described in our documentation, that you can find [here](https://docs.medusajs.com/plugins/payment/stripe).

**Comment by olivermrbl** [comment]
Nice catch @kasperkristensen. I didn't notice the `pk_` in the message at first glance, but this explains the issue and confirms my suspicion. 

I'll be closing this issue for now, @Sylvesterx1.

I would like to kindly ask you to first search in our documentation when you are experiencing issues. Often times you'll find the answer there or alternatively seek help in our Discord.

**Comment by abhisaara** [comment]
in Medusa 1 it still not working

**Comment by martinmakejev** [comment]
Hi, did you manage to fix it and how? On Medusa 1.20 version i face the same problem using Stripe test version. I did everything like in the documentation were described, tried to upgrade and downgrade Medusa version's, still no success.

**Comment by ofoetey** [comment]
@martinmakejev ensure that you are using the publishable key "pk_*" for the storefront and "sk_*" for the backend
